development/tools/pkg/githubrelease/release.go

```diff
@@ -1,105 +1,104 @@
 package githubrelease
 
 import (
+	"context"
 	"io/ioutil"
 	"os"
-	"path"
 
+	"github.com/google/go-github/github"
 	"github.com/pkg/errors"
 )
 
-// Release .
-type Release struct {
-	Github  *GithubAPIWrapper
-	Storage *StorageAPIWrapper
-}
+const tmpFilePattern = "release-artifact-"
 
-var kymaArtifactsDir = "kyma-artifacts"
+// releaseWizard provides functions to create a complete Github Release
+type releaseWizard struct {
+	github  GithubAPI
+	storage StorageAPI
+}
 
-//CreateRelease .
-func (gr *Release) CreateRelease(releaseVersion string, targetCommit string, releaseChangelogName string, localArtifactName string, clusterArtifactName string, isPreRelease bool) error {
-	artifactsDir, err := ioutil.TempDir("", kymaArtifactsDir)
-	if err != nil {
-		return errors.Wrapf(err, "while creating temp %s dir", kymaArtifactsDir)
+// NewReleaseWizard returns new instance of releaseWizard
+func NewReleaseWizard(github GithubAPI, storage StorageAPI) releaseWizard {
```

Comment:
Try to avoid returning private structures from functions. Function caller is able to use it only in current scope, but is not able to pass it to other function or return. If you want to keep this structure private I'd rather hide it behind interface.

Reply:
done